examples/C++/DeadlineQoSExample/CMakeLists.txt

```diff
@@ -37,7 +37,11 @@ file(GLOB DEADLINEQOS_EXAMPLE_SOURCES "*.cxx")
 
 add_executable(DeadlineQoSExample ${DEADLINEQOS_EXAMPLE_SOURCES})
 target_include_directories(DeadlineQoSExample PRIVATE ${Boost_INCLUDE_DIR})
-target_link_libraries(DeadlineQoSExample fastrtps fastcdr ${Boost_LIBRARIES})
+if(UNIX)
+  target_link_libraries(DeadlineQoSExample fastrtps fastcdr ${Boost_LIBRARIES} pthread)
+else()
+  target_link_libraries(DeadlineQoSExample fastrtps fastcdr ${Boost_LIBRARIES})
+endif()
```

Comment:
To avoid duplicating all this logic I would suggest to update the patch to:

```
target_link_libraries(DeadlineQoSExample fastrtps fastcdr ${Boost_LIBRARIES})
if(UNIX)
  target_link_libraries(DeadlineQoSExample pthread)
endif()
```